### ISSUE TITLE
Set up Google Doc template with Google Apps Script attach

### ISSUE THREAD

**Issue by TylerFisher** [news-catalyst/tiny-news-demo#30]

**Comment by jacqui** [comment]
Apparently, I need to switch the kind of project I'm using in Google Drive to do this first... on it now, and I barely know what's going on 🤪

> User-managed Cloud Platform project required
> You're using an Apps Script-managed Cloud Platform project. In order to publish, you'll need to switch to a user-managed Cloud Platform project for this script.
> 

 https://developers.google.com/apps-script/guides/cloud-platform-projects#switch_to_a_different_google_cloud_platform_project

**Comment by jacqui** [comment]
Jeesh, this is complicated and confusing, but I've been making progress. I'm basically following along with these instructions: https://developers.google.com/gsuite/add-ons/how-tos/publishing-editor-addons

Seems we have to publish the add-on to get it available in documents. Fine. It's not doing anything really proprietary or complicated. 

There are many steps required to publish a script, though.

1. I've created a Google Cloud Platform project: [tiny news co demo](https://console.cloud.google.com/iam-admin/settings?authuser=1&organizationId=353951357297&project=chromatic-tree-278523) because the script must be associated with a "user managed project"
2. I've enabled the "G Suite Marketplace SDK" under APIs & Services in the GCP
3. I've filled in an Oauth Consent Screen
4. I've filled in a configuration form for G Suite Marketplace SDK
5. I've gone through the [development checklist](https://developers.google.com/gsuite/add-ons/how-tos/publishing-editor-addons#development_checklist) and added comments, updated code to follow the instructions.
6. I've ensured the script is requesting and using the right Oauth scopes for working with documents.
7. I've [created a first version of the script](https://developers.google.com/apps-script/guides/versions#creating_a_version) and just called it version `1`
8. I made a quick icon for the script because it's required as well (it's a super tiny house)

Now we have to wait: the add-on [requires review by Google](https://developers.google.com/gsuite/add-ons/concepts/editor-addon-review) before it can be published.

I found this process pretty hard to follow. I did initially try several approaches to including the script in the template but was stymied. If you know how to do this, @TylerFisher, I'm all ears. I could get the add-on menu to show up (`Embed`) in the template document (see below), but it doesn't display in documents I create from the template. This is how I ended up going through all the hoops to publish the add-on.

<img width="1176" alt="Screen Shot 2020-05-27 at 2 25 55 pm" src="https://user-images.githubusercontent.com/3397/82977758-26e58980-a026-11ea-8dce-4cf7f8fb6491.png">

Here's a [link to the script](https://script.google.com/a/newscatalyst.org/d/13-DrK3l4Hv8_dU2589vIsC1qs2rM7d9iLsMGGAxv_eL3Zuvfa2PL_Ygr/edit?splash=yes)

**Comment by jacqui** [comment]
So I decided to revisit this today. I really would love to wrap this bit of work up (so say we all). 

I ended up finding this guide - [Install G Suite Marketplace apps in your domain](https://support.google.com/a/answer/172482?hl=en) - which seems like a way forward. 

However, I got stuck on step 1:

```
Your current account, jacqui@newscatalyst.org, doesn't have permission to do these steps. To continue, switch to an administrator account. This will open the Google Admin console.
```

@TylerFisher are you an admin of our G Suite organization account? I just made sure to share the app script with you (tyler@newscatalyst.org). You should be able to view it at https://script.google.com/u/1/home. Can you make me an admin? Thanks!

**Comment by jacqui** [comment]
So... I think we've decided that we're not pursuing this any longer, and that to embed content one will do so by adding a URL on its own line in the google doc. We'll provide instructions on how to do this.

Slight update though, just to make sure the current status of the Google Apps Script saga is noted: I got the embed script installed for our domain. However I can't actually add it to a google doc 😡. Screenshots below; it says it is installed, but the menu doesn't show up. 

<img width="1296" alt="Screen Shot 2020-06-17 at 1 10 15 pm" src="https://user-images.githubusercontent.com/3397/84850975-7d316f80-b09c-11ea-89dc-df786e620f18.png">

<img width="1406" alt="Screen Shot 2020-06-17 at 1 10 37 pm" src="https://user-images.githubusercontent.com/3397/84851033-9df9c500-b09c-11ea-9717-e62816511e71.png">

I'm closing this issue. We can revisit the idea of adding a custom menu with behaviour to google docs in the future, but given how unpredictable this process and implementation seems to be, I don't look forward to us trying again 😕🤞
